packages/framework/src/Model/Product/Search/ProductElasticsearchRepository.php

```diff
@@ -242,7 +242,7 @@ public function deletePresent(int $domainId, array $deleteIds): void
                     'bool' => [
                         'must' => [
                             'ids' => [
-                                'values' => $deleteIds,
+                                'values' => array_values($deleteIds),
```

Comment:
Why is this necessary?

Reply:
Because when you sent array indexed somehow different than from zero, it’s translated as an object not an array and Elasticsearch fails. 
For example this code is not really expected to fail 
```php
$productIds = [5,6,7];
$doNotUpdateIds = [5];
$toUpdateIds = array_diff($productIds, $doNotUpdateIds);

$elasticRepository->delete($toUpdateIds);